### PR TITLE
Test Case implementation

### PR DESCRIPTION
**Pull request by prklm10**
Test Case implementation
Adding support for same snapshot name across different test-cases.

### CODE REVIEW

File: packages/core/src/discovery.js [hunk 1]
```diff
@@ -317,7 +317,7 @@ export function createDiscoveryQueue(percy) {
     .handle('error', ({ name, meta }, error) => {
       if (error.name === 'AbortError' && queue.readyState < 3) {
         // only error about aborted snapshots when not closed
-        percy.log.error('Received a duplicate snapshot, ' + (
+        percy.log.error('Received a duplicate snapshot within same test case, ' + (
```
Comment (ninadbstack) [inline packages/core/src/discovery.js:320]: not everyone would be using testcases, so the wording should depend on if testCase was null or not

File: packages/core/test/snapshot.test.js [hunk 2]
```diff
@@ -837,14 +837,62 @@ describe('Snapshot', () => {
...
+      'dom-snapshot': testDOM
+    }, {
+      url: 'http://localhost:8000/one',
+      testCase: 'test-case-4',
+      domSnapshot: JSON.stringify({ html: testDOM })
+    }]);
+
+    expect(logger.stderr).toEqual([]);
+    expect(logger.stdout).toEqual([
+      '[percy] Snapshot taken: /one',
```
Comment (ninadbstack) [inline packages/core/test/snapshot.test.js:889]: we need to update logging and `meta` to include testcase name

Comment (prklm10) [reply packages/core/test/snapshot.test.js]: updated meta as well as logging.

File: packages/core/src/discovery.js (hunk 1 above)
Comment (prklm10) [reply packages/core/src/discovery.js]: Fixed logging

**Pull request merged**